### DISCUSSION TITLE
Computed route params are updated before page transition

### DISCUSSION THREAD

**Discussion by johannesstricker** [nuxt/nuxt#16022] [Questions]
I have a problem when transitioning from one page to the next. I'm using computed route params to update my data when the params change, e.g.
```
const route = useRoute();
const slug = computed(() => route.params.slug);
const { data, refresh } = useAsyncData(...);
watch(slug, refresh);
```

When I navigate to another page (by clicking on a `NuxtLink`), the computed slug will be `undefined` for a brief moment, because the route already changed, but the page is not yet loaded.
This leads to another network request and causes my current page, the one I'm transitioning away from, to rerender without any data, before the new page is shown. How do I properly avoid that?

**Comment by johannesstricker** [comment]
Maybe this is related to https://github.com/vuejs/core/issues/2291?

**Reply by johannesstricker** [reply]
The fix mentioned [here](https://github.com/vuejs/core/issues/2291#issuecomment-705271501) by Evan You does make things a little better, I believe, but it still doesn't fix the problem. It seems that while `useAsyncData` runs the old page is still visible, but the slug has already been updated, leading to new data being fetched.

**Reply by lacorde** [reply]
Hi @johannesstricker, did you ever find a workaround for this issue ?

**Reply by johannesstricker** [reply]
Not really, but I haven't looked into it recently.
